### PR TITLE
XRT-517 driver cleanup: sudev lock, dev instance, p2p lock

### PR DESCRIPTION
**Pull request by houlz0507**

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c [hunk 1]
```diff
@@ -467,8 +471,13 @@ static int __xocl_subdev_create(xdev_handle_t xdev_hdl,
 
 	subdev->pldev->dev.parent = &core->pdev->dev;
 
+	/* lock dev, no offline, no destroy */
+	subdev->hold = true;
+	xocl_unlock_xdev(xdev_hdl);
+
 	retval = platform_device_add(subdev->pldev);
 	if (retval) {
+		xocl_lock_xdev(xdev_hdl);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c:480]: don't need to unhold dev?

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c [hunk 2]
```diff
@@ -488,13 +497,16 @@ static int __xocl_subdev_create(xdev_handle_t xdev_hdl,
 	 */
 	retval = device_attach(&subdev->pldev->dev);
 	if (retval != 1) {
+		xocl_lock_xdev(xdev_hdl);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c:500]: don't need to unhold?

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c [hunk 3]
```diff
@@ -1090,6 +1092,8 @@ int xocl_p2p_init(struct xocl_dev *xdev)
 	xocl_info(&pdev->dev, "Initializing P2P, bar %d, len %lld",
 			xdev->p2p_bar_idx, xdev->p2p_bar_len);
 
+	mutex_init(&xdev->p2p_mem_chunk_lock);
```
[the remark this answers is not in the document]
Comment (houlz0507) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c]: I moved lock init() and destroy() to userpf_probe() and userpf_remove().  Then we will never worry about init/destroy this lock

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c (hunk 1 above)
Comment (houlz0507) [reply src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c:480]: added. Thanks.

**Pull request merged**